src/lib/log.c

```diff
@@ -19,19 +19,94 @@
 
 #define _GNU_SOURCE
 
+#include <ctype.h>
 #include <stdarg.h>
+#include <stdlib.h>
+#include <string.h>
 #include <time.h>
 
 #include "log.h"
 #define DEBUG_HEADER_SIZE 50
 
 static int cur_log_level = LOG_INFO;
+static int json_format = 0;
 
 void set_log_level(int log_level)
 {
 	cur_log_level = log_level;
 }
 
+void set_json_format(void)
+{
+	json_format = 1;
+}
+
+void json_start(const char *op)
+{
+	if (json_format) {
+		fprintf(stderr, "[\n");
+		fflush(stderr);
+		fprintf(stderr, "{ \"%s\" : \"start\" },\n", op);
```

Comment:
we could use the same format here:

    {type: "start", section: "bundle-add" }

Reply:
good idea